apps/channels/audio.py

```diff
@@ -17,6 +20,60 @@ def get_transcript(audio: BytesIO) -> str:
 
 
 def synthesize_voice(text: str, synthetic_voice: SyntheticVoice) -> Tuple[BytesIO, float]:
+    if synthetic_voice.service == "AWS":
+        return AWS_synthesize_voice(text, synthetic_voice)
+    elif synthetic_voice.service == "azure":
+        return azure_synthesize_voice(text, synthetic_voice)
+
+    raise AudioSynthesizeException(f"Unrecognized audio service: {synthetic_voice.service}")
+
+
+def azure_synthesize_voice(text: str, synthetic_voice: SyntheticVoice) -> Tuple[BytesIO, float]:
+    """
+    Calls Azure's cognitive speech services to convert the text to speech using the synthetic_voice
+    """
+    try:
+        speech_config = speechsdk.SpeechConfig(
+            subscription=settings.AZURE_SUBSCRIPTION_KEY, region=settings.AZURE_REGION
+        )
+
+        temp_file = tempfile.NamedTemporaryFile(delete=False, suffix=".wav")
+
+        # Create an audio configuration that points to the output audio file
+        audio_config = speechsdk.audio.AudioConfig(filename=temp_file.name)
+
+        # Setup voice font (Azure refers to synthetic voices as voice fonts)
+        speech_config.speech_synthesis_voice_name = f"{synthetic_voice.language_code}-{synthetic_voice.name}"
+
+        # Create a speech synthesizer
+        synthesizer = speechsdk.SpeechSynthesizer(speech_config=speech_config, audio_config=audio_config)
+
+        # Synthesize the text
+        result = synthesizer.speak_text_async(text).get()
```

Comment:
Should this operation not be blocking i.e. use `speak_text` instead?

Reply:
No idea. I changed it though https://github.com/dimagi/open-chat-studio/pull/48/commits/28a18f1a98efbdae682b2ee8a3b6bd02a170e5d3